code/workspaces/infrastructure-api/src/controllers/stackController.js

```diff
@@ -113,10 +114,23 @@ const createStackValidator = [
       }
       return true;
     }),
+  check('visible', 'visible must be specified for sites')
+    .custom((value, { req }) => {
+      if (indexOf(Stacks.getNamesByCategory(PUBLISH), req.body.type) > -1) {
```

Comment:
I think you can do this slightly more transparent way now using `.includes`. Something like 
```js
if (Stacks.getNamesByCategory(PUBLISH).includes(req.body.type)) {
  ...
}
```
(I've just noticed you have used `.includes` on the next line!)